### PR TITLE
[Dashboard]: Create new team card

### PR DESCRIPTION
**Pull request by Nortsova**
## Description

Story
As a USER, I want to easily create new teams from the dashboard to better compartmentalize my colony’s workflow.
Description
When a colony is freshly created, as there are no other teams available, we will display a card for 'Create new team' action, which will open the side action panel opens with the Create team action type pre-selected.

- [ ] For colonies not having any teams yet, a 'Create new team' card is shown

new-colony-ui

## Testing
* Step 1. Run an app
* Step 2. Run `node scripts/create-colony-url.js`
* Step 3. Copy link and create new colony
* Step 4. In new colony you will see an empty new card component.
* Step 5. Click on "+ New Team" item
* Step 5. "New Team" motion should be opened
new-colony-ui

## TODO

- [ ] This PR is just the beginning; the remaining functionality will be addressed in #2851
- [ ] Refactor or remove the WidgetBox component, as it has similar functionality to WidgetCard and will be partially removed during the dashboard feature work.

Contributes to #2852

### CODE REVIEW

File: src/components/v5/common/WidgetCards/partials/TitleComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { titleClassName } from '../styles.ts';
+
+interface TitleProps {
+  title: React.ReactNode;
+}
+
+export const TitleComponent: FC<TitleProps> = ({ title }) => (
+  <span>
+    <h3 className={titleClassName}>{title}</h3>
+  </span>
```
Comment (rumzledz) [inline src/components/v5/common/WidgetCards/partials/TitleComponent.tsx:12]: Is there an advantage to nesting `h3` within `span` here?

File: src/components/v5/frame/ColonyHome/partials/FundsCards/FundsCards.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const { colony } = useColonyContext();
+
+  const teams = colony.domains?.items || [];
+
+  return (
+    <WidgetCards.List>
+      {teams.map((item) => {
+        return <WidgetCards.Item title={item?.metadata?.name} />;
+      })}
+      {/* Teams always will have at least 1 team by default - it is general */}
```
Comment (rumzledz) [inline src/components/v5/frame/ColonyHome/partials/FundsCards/FundsCards.tsx:31]: 👌

**Review by rumzledz — CHANGES_REQUESTED** [review]
Awesome work on this @Nortsova ! I can confirm that this new Create Team button is present for a colony that doesn't have a team yet. And when clicked, it brings up the Action Form and sets the Action type correctly 💯 



Although I did notice that the design is slightly off on mobile but I'm not sure if this fix must be done under this PR. The width isn't filling up the horizontal space basically:

| Figma | This PR |
| ---- | ---- |
| Screenshot 2024-08-30 at 06 32 44 | Screenshot 2024-08-30 at 06 31 51 |

If it's something that's already on `feat/enhanced-dashboard`, just ignore this 🙅 

Then I left a little comment about the `h3` component nested within `span`. If that's really needed due to styling etc. just let me know and I'll approve it 👌

File: src/components/v5/common/WidgetCards/partials/TitleComponent.tsx (hunk 1 above)
Comment (Nortsova) [reply src/components/v5/common/WidgetCards/partials/TitleComponent.tsx]: Thanks for noticing that, regarding mobile position I will handle it as part of #2851 , thanks!

**Review by rumzledz — APPROVED** [review]
LGTM @Nortsova !! 🚀

File: src/components/v5/common/WidgetCards/WidgetCardsItem.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  variant?: WidgetCardsItemVariant;
+  icon?: Icon;
+  title?: React.ReactNode;
+  onClick?: () => void;
+  className?: string;
+}
+
+const variantClassNames: Record<WidgetCardsItemVariant, string> = {
+  default: '',
+  dashed: 'border-dashed',
```
Comment (bassgeta) [inline src/components/v5/common/WidgetCards/WidgetCardsItem.tsx:20]: do we need to use the `tw` util here for code completion?

File: src/components/v5/common/WidgetCards/WidgetCardsItem.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { type Icon } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import React, { type FC, type PropsWithChildren } from 'react';
+
+import { ContentComponent } from './partials/ContentComponent.tsx';
```
Comment (bassgeta) [inline src/components/v5/common/WidgetCards/WidgetCardsItem.tsx:5]: maybe something like `WidgetCardContent` would be be a bit less generic :D

File: src/components/v5/common/WidgetCards/styles.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { tw } from '~utils/css/index.ts';
+
+export const wrapperClassName = tw`flex min-h-[7.4rem] min-w-[16.5rem] items-center justify-between gap-2 rounded-lg border px-6 py-6`;
```
Comment (bassgeta) [inline src/components/v5/common/WidgetCards/styles.ts:3]: do all of these need to be in a separate file? we can have them inside the component className, unless we want something to do with them later on

**Review by bassgeta — CHANGES_REQUESTED** [review]
Nice work on this, the code is very readable and it's composed quite nicely.
I've left some comments regarding naming and those styles in a separate file, which we can discuss and change/just resolve the comments :D

Functionality works:
image
image
Mobile:
image
Sidebar opens :heavy_check_mark: 
image

**Review by mmioana — APPROVED** [review]
Awesome work @Nortsova! Thank you for picking up this issue 🌟 

We now have a `Create team` card for a freshly created colony 🎉 

https://github.com/user-attachments/assets/91918ceb-3821-4e07-a1bf-4631a48a7b3c

I will approve it, but would be nice to tackle Kerry's suggestions 💯

**Review by bassgeta — APPROVED** [review]
Still looking good, thanks for the changes :ok_hand: We can merge this badboi now!

**Pull request merged**